### DISCUSSION TITLE
Experimental release with Neural Amp Modeler plugin

### DISCUSSION THREAD

**Discussion by djshaji** [djshaji/amp-rack#67] [Announcements]
This build contains the [LV2 Neural Amp Modeler plugin](https://github.com/djshaji/neural-amp-modeler-lv2-amprack) adapted from the [LV2 plugin](https://github.com/mikeoliphant/neural-amp-modeler-lv2) by [Mike Oliphant](https://github.com/mikeoliphant)

It doesn't work properly on my devices :disappointed: 

Please check and report if it works on your phone.

You can download a model from here:
https://tonehunt.org/all

<hr /><em>This discussion was created from the release <a href='https://github.com/djshaji/amp-rack/releases/tag/4.05'>Experimental release with Neural Amp Modeler plugin</a>.</em>

**Comment by piedpipr** [comment]
The plugin is apparently working it's just the audio that's clipping with noticeable latency. I have some suggestions to find out the actual reason.

1. Try increasing the buffer sample at to least 128 or more if necessary. Even if it introduces more latency if the sound is working as expected it all comes down to the raw processing power of the device and host in relationship to latency.

2. Try AIDA DSP's AIDA-X as a light alternative.

3. Apparently some noise are due to lack of gain control and gate. 

( as of now it seems even my Pixel 6a is underpowered for NAM. But I don’t believe so if there’s any other issue like x86 vs ARM architecture problem. Again MOD has their own NAM LV2 running on a stand alone device)

**Reply by piedpipr** [reply]
The latest release says 'No such file found' while loading NAM models. Otherwise it's working fine.

**Comment by arutemu** [comment]
Agreed with the previous commenter, the plugin itself seems to work, right until you try to load the NAM model file.
Right after choosing the model file the app crashes.
Tried with different models

Device: Samsung Galaxy Tab S8.

Adding some logs I was able to extract.
[amprack_log.txt](https://github.com/djshaji/amp-rack/files/15049641/amprack_log.txt)

**Comment by arutemu** [comment]
I've just checked the 4.07 version,  unfortunately the app still crashes at the same point. 
The NAM files were extracted from the archive beforehand.

**Reply by piedpipr** [reply]
Crashing straight away. 

![Screenshot_20240421-204750.png](https://github.com/djshaji/amp-rack/assets/43669876/30565d4b-ccd4-4374-b569-67a9877a9578)

**Comment by djshaji** [comment]
Hi, try this build:
https://github.com/djshaji/amp-rack/releases/tag/4.07a

Apparently (I think) the issue might be due to file extensions. I have not added a _default_ for everything that is not a zip file to be considered as a nam model. Android it seems identifies mime types based on extensions.

**Comment by arutemu** [comment]
@djshaji I am happy to report that ver. 4.07a is working.
I was able to load a few different models and the plugin generates different types of sound 👍 

There are 2 small usability issues that you may improve in the future.
1) The names of all loaded models consist of numbers only.
It would be great if you could import the model name for good visibility.
2) If possible, it would be nice to have an option to load a bunch of models at once.

![image](https://github.com/djshaji/amp-rack/assets/6502628/a352f86d-ba39-4d4d-9b05-226ec6d2ae1d)

![Screenshot_20240421_181243_Files](https://github.com/djshaji/amp-rack/assets/6502628/4d595a60-f517-4f6b-9761-462321f419d9)

**Reply by djshaji** [reply]
Hi, I've published a new build that uses base names of selected files instead of numbers. Also, you can directly load zip files with nam models, and the app will unzip and add models from that automatically.

[Build 4.08](https://github.com/djshaji/amp-rack/releases/tag/4.08
)

Also I've been trying to add impulse response plugins. I've compiled the TAP IR plugin successfully, and it runs too. But it doesn't work :disappointed: 

I think one big reason for that is that I've not implemented the LV2 specification completely. The app doesn't support atoms, log, file and schedule extensions, which many plugins use. I've been basically using workarounds instead, which hasn't been working out too well.

So I've finally decided to implement LV2 extension specifications completely. It's gonna be difficult, but in the long run it will be worth it, and if it is done, then adding new plugins will be simply a matter of compiling and adding the shared libraries to the jniLibs folder.

I also want to write documentation about the entire project structure, the design decisions, why I designed it the way I did,. and most importantly, how to add new plugins. My wish is that when I die, the app should live on. 

Also, i just found out about another open source plugin specification: [CLAP](https://u-he.com/community/clap/)

Since the plugins are also written in C, I think it may :crossed_fingers: be possible to add support for them

**Reply by arutemu** [reply]
Thank you for the update, Mr. Shaji!
I happened to be online and am already downloading the update (to look for old and new bugs, hehe)

I am happy to hear about the work done, and at the same time sad to hear about the difficulties in implementing TAP IR plugin.
But I would advise separating different topics into different threads/discussions, to keep all relevant talks in a proper places :D

During the last week I found a few more UI issues, so if they are still present in the new build - will update it with more details.

**Reply by arutemu** [reply]
And one more update - the NAM model names are much more readable now :)
And loading multiple models from a single Zip file also works properly.
![Screenshot_20240503_114127_Amp Rack](https://github.com/djshaji/amp-rack/assets/6502628/3d4a61d3-d72c-40d4-96de-fb55566cd96e)

**Comment by joebowbeer** [comment]
Plans to support NAM A2?

https://www.neuralampmodeler.com/post/architecture-a2
